### PR TITLE
Provide more details on date parsing

### PR DESCRIPTION
**Pull request by jskeet**

### CODE REVIEW

File: src/NodaTime.Web/Markdown/3.0.x/localdate-patterns.md [hunk 1]
```diff
@@ -208,3 +208,25 @@ the inclusion of a leap month, the month name/number correspondence changes in a
 Until this is fixed, it is strongly recommended that you only use month *numbers* in any textual
 representations of dates in the Hebrew calendar. Additionally, you may wish to consider how to
 best clarify whether that month number is in the scriptural or civil numbering system.
+
+**Parsing and dependent values**
+
+Text parsing consists of two phases.
+
+The first phase obtains values from the text independently.
```
Comment (malcolmr) [inline src/NodaTime.Web/Markdown/3.0.x/localdate-patterns.md:216]: Maybe "parses values from the text independently, left-to-right"?

File: src/NodaTime.Web/Markdown/3.0.x/localdate-patterns.md [hunk 2]
```diff
@@ -208,3 +208,25 @@ the inclusion of a leap month, the month name/number correspondence changes in a
...
+These values cannot affect each other during the first phase, which leads to a small limitation
+in date patterns: a date pattern cannot contain both a calendar specifier (`c`) and an era specifier (`g` or `gg`). The parser
+would not be able to determine which eras are valid without taking the calendar into account.
```
Comment (malcolmr) [inline src/NodaTime.Web/Markdown/3.0.x/localdate-patterns.md:219]: "which eras were valid"? (I'm not sure myself)

File: src/NodaTime.Web/Markdown/3.0.x/localdate-patterns.md [hunk 3]
```diff
@@ -208,3 +208,25 @@ the inclusion of a leap month, the month name/number correspondence changes in a
...
+
+The second phase combines and validates values. For example, a "day of month" value may be valid in
+some year/month/calendar combinations but not in others. Likewise, a "day of week" value can only be validated
+when the complete date is known.
+
+The ordering of these phases can lead to some potentially-unexpected results in the case of errors.
+For example, parsing the text "2000-99-00" with an ISO date rule, the result will be a failure
```
Comment (malcolmr) [inline src/NodaTime.Web/Markdown/3.0.x/localdate-patterns.md:226]: "For example, when attempting to parse the text..."?

(or maybe "when parsing..." or "in parsing..."?)

**Review by malcolmr — APPROVED** [review]
LGTM

Thanks, this looks nice and clear.

The location seems reasonable too, since this really does only affect date parsing (I had wondered if we should put this text somewhere more general, but I think this is okay as-is).

File: src/NodaTime.Web/Markdown/3.0.x/localdate-patterns.md (hunk 1 above)
Comment (jskeet) [reply src/NodaTime.Web/Markdown/3.0.x/localdate-patterns.md:216]: Done.

File: src/NodaTime.Web/Markdown/3.0.x/localdate-patterns.md (hunk 2 above)
Comment (jskeet) [reply src/NodaTime.Web/Markdown/3.0.x/localdate-patterns.md:219]: I'm on the fence too, but have gone with were.

File: src/NodaTime.Web/Markdown/3.0.x/localdate-patterns.md (hunk 3 above)
Comment (jskeet) [reply src/NodaTime.Web/Markdown/3.0.x/localdate-patterns.md:226]: Gone with "when parsing".

**Pull request merged**